lib/internal/async_local_storage/async_context_frame.js

```diff
@@ -53,12 +54,15 @@ class AsyncLocalStorage {
   }
 
   run(data, fn, ...args) {
-    const prior = AsyncContextFrame.current();
+    const prior = this.getStore();
+    if (ObjectIs(prior, data)) {
```

Comment:
Do we expect NaN and -0 to be passed through? I would just use a regular strict equal comparison. That is much faster than using Object.is().

Reply:
Not really because of `NaN` and `-0`.
I made it like in the other ALS implementation (see [here](https://github.com/nodejs/node/blob/c46b2b9da30628585900b53052aec509cb59036e/lib/internal/async_local_storage/async_hooks.js#L104-L107))

I'm fine with using `===` but honestly I wonder why v8 is unable to optimize the diff away.

Another option would be to remove the check at all.

Reply:
Makes sense. I guess we could change those as well. I believe the semantics are always more difficult for V8 due to being a function call and does not allow to optimize the code as much due to potential side effects in case it is patched (even if it's just that check).